core/insertion_marker_manager.js

```diff
@@ -617,8 +624,21 @@ InsertionMarkerManager.prototype.showInsertionMarker_ = function() {
   const closest = this.closestConnection_;
 
   const isLastInStack = this.lastOnStack_ && local === this.lastOnStack_;
-  const imBlock = isLastInStack ? this.lastMarker_ : this.firstMarker_;
-  const imConn = imBlock.getMatchingConnection(local.getSourceBlock(), local);
+  let imBlock = isLastInStack ? this.lastMarker_ : this.firstMarker_;
+  let imConn;
+  try {
+    imConn = imBlock.getMatchingConnection(local.getSourceBlock(), local);
+  } catch (e) {
+    // It's possible that the number of connections on the local block has
+    // changed since the insertion marker was originally created.  Let's recreate
+    // the insertion marker and try again.
+    // In theory we could probably recreate the marker block (e.g. in getCandidate_),
```

Comment:
+1 to only recreating the insertion marker if something has actually changed, since building a block can be slow.

Reply:
@rachel-fenichel Just to be clear, I think that you're responding to my comment in the code and saying that you're ok with the way that I did it, as opposed to the option that I mentioned in the comment.  Do I have that right?